bikespace_frontend/src/components/submission/location/Location.tsx

```diff
@@ -1,41 +1,68 @@
-import React from 'react';
+import React, {RefObject, useRef, useEffect} from 'react';
+import MapGL, {
+  Marker,
+  AttributionControl,
+  GeolocateControl,
+} from 'react-map-gl/maplibre';
+
 import {LocationLatLng} from '@/interfaces/Submission';
-import {MapContainer, TileLayer, Marker} from 'react-leaflet';
-import {LatLngTuple} from 'leaflet';
 
-import 'leaflet/dist/leaflet.css';
-import 'leaflet-defaulticon-compatibility';
-import 'leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css';
+import 'maplibre-gl/dist/maplibre-gl.css';
 
 import styles from './location.module.scss';
 
 export interface LocationProps {
   location: LocationLatLng;
-  handler: React.ReactNode;
+  setLocation: React.Dispatch<React.SetStateAction<LocationLatLng>>;
 }
 
-function Location({location, handler}: LocationProps) {
-  const position = [location.latitude, location.longitude] as LatLngTuple;
+function Location({location, setLocation}: LocationProps) {
+  const geoControlRef =
+    useRef<maplibregl.GeolocateControl>() as RefObject<maplibregl.GeolocateControl>;
+
+  const {latitude, longitude} = location;
+
+  useEffect(() => {
+    if (!geoControlRef.current) return;
+
+    geoControlRef.current.trigger();
+  }, [geoControlRef.current]);
```

Comment:
See comment on `SubmissionForm.tsx` - I think it's cleaner to load the user's location before rendering this component instead of using the geolocate control to do it when the map loads.

One other thing I noticed when I was testing it was that the geolocate control didn't trigger until I clicked the map (which is weird, because this is how the react-map-gl docs say to do it). Putting `    geoControlRef.current.trigger();` in the `onLoad` method of the `MapGL` component does work as intended though.